frontend/src/stores/assignment.ts

```diff
@@ -0,0 +1,103 @@
+import { reactive } from 'vue';
+import { defineStore } from 'pinia';
+
+import { assignmentTemplate } from '@/helpers/assignmentModal';
+import {
+  type IAssignment,
+  type IContacts,
+  type IPhoneNumbers,
+  type IAddresses
+} from '@/utils/interfaces';
+
+type TypeAssignmentKeys = keyof IAssignment;
+
+export const useAssignmentStore = defineStore('assignment', () => {
+  const assignment: IAssignment = reactive(
+    JSON.parse(JSON.stringify(assignmentTemplate))

```

Comment:
if you need to deeply clone an object use structuredCopy

Reply:
Done